### PR TITLE
DBZ-1497 Using int64 for INTERVAL microseconds mapping

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-1497

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/time/MicroDuration.java [hunk 1]
```diff
@@ -8,26 +8,25 @@
...
+
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.SchemaBuilder;
 
 /**
- * A utility representing a duration into a corresponding {@link SchemaBuilder#float64() FLOAT64} 
+ * A utility representing a duration into a corresponding {@link SchemaBuilder#int64() INT64}
  * number of <em>microsecond</em>, and for defining a Kafka Connect {@link Schema} for duration values.
- * 
- * The duration is amount is stored in floating point as opposed to an integer because certain DBs (e.g. Postgres) will store
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/time/MicroDuration.java]: @jpechane, this comment makes the original motivation for using float clear: it's indeed the larger value range. I'm still inclined to prefer the int64 mapping, as the fractional part just doesn't make sense. In that light, we may do the String-based mapping ([DBZ-1498](https://issues.jboss.org/browse/DBZ-1498) sooner than later, though (e.g. "1Y2M3DT4H5M6.78S").

File: debezium-core/src/main/java/io/debezium/time/MicroDuration.java [hunk 2]
```diff
@@ -71,16 +70,16 @@ private MicroDuration() {
      * @param micros a number of microseconds
      * @param daysPerMonthAvg an optional value representing a days per month average; if null, the default duration
      * from {@link ChronoUnit#MONTHS} is used.
-     * @return a {@link BigDecimal} value which contains the number of microseconds, never {@code null}
+     * @return @return Approximate representation of the given interval as a number of microseconds
     */
-    public static double durationMicros(int years, int months, int days, int hours, int minutes, double seconds,
+    public static long durationMicros(int years, int months, int days, int hours, int minutes, double seconds,
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/time/MicroDuration.java]: @gunnarmorling Why are seconds kept double?

**Comment by gunnarmorling** [comment]
@jpechane, you had asked somewhere why the `seconds` parameter still is a double. The reason is that this is what we get from PG (see `PostgresValueConverter.convertInterval()`. Now I could split that value into two longs for second and microsecond part, but I figured it'd be easier to just pass on the double.

**Comment by jpechane** [comment]
@gunnarmorling Thanks for explanantion, I've found it in the docs just a minute after I sent the comment so I deleted it. Sorry for the confusion. Albeit thinking about it this is of course not correct as some microsecond values will not be represented precisely. And as double comes from JDBC it does not matter if you use it or not, the precision is lost.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks!

**Pull request merged**